Heroku

JPrevost opened the pull request:
Adds some common practices with our Heroku deployed apps and links
to relevant upstream documentation.

Documents the weirdness that is the generated sidenav for this documentation site.
Basically, it's now set to fallback to alphabetical order except for the "readme / home"
page.

closes #4

JPrevost committed:
Sidenav instructions

JPrevost committed:
Adds heroku documentation

Adds some common practices with our Heroku deployed apps and links
to relevant upstream documentation.

thatandromeda commented on heroku.md:
```diff
@@ -0,0 +1,182 @@
+## Pipelines
+
+TL;DR: Use [pipelines](https://devcenter.heroku.com/articles/pipelines).
+They are :mega: :rainbow: :beers: :fireworks:
```
📣 🌈 🍻 🎆

thatandromeda commented on heroku.md:
```diff
@@ -0,0 +1,182 @@
+normally a quick blip and doesn’t require downtime announcements.
+
+
+## Production Deploy with Pipelines Cheatsheet
```
👍

thatandromeda commented on heroku.md:
```diff
@@ -0,0 +1,182 @@
+
+### Confirm the app restarts and seems good
+It takes a few seconds to restart. If things are _not_ good, rollback
+in the Dashboard.
```
Does this only roll back the codebase, or does it also roll back the database? (i.e. are there additional steps that need to be taken if the deploy involved a database schema migration)

JPrevost replied:
Righto. I'll note that. It's just env and code. Any db you screwed with is your problem. Good call to note that.

thatandromeda commented on heroku.md:
```diff
@@ -0,0 +1,182 @@
+
+## Viewing / changing Environment Variables
+
+Using ENV for config is common for Heroku deploys. You can view and
```
Do/should we have a policy about keeping sensitive ENV variables in the shared notes folder in lastpass?

JPrevost replied:
We do not. We should. I'll note that it should be done here as clearly it should be done. 👍

thatandromeda commented on heroku.md:
```diff
@@ -0,0 +1,182 @@
+builds. Using a Moira list with an email hook allows for interested
+stakeholders to stay in the loop (note: keep in mind the
+deploy hooks don't always have a ton of context as to what has
+changed so only alerting truly engaged stakeholders with this
```
Not sure 'engaged' is the right word here...business owners, project managers, etc. might be very *engaged* but also not get any useful content from the deploy hook messages. "..., so only alert people who will get mileage out of these messages"? It might help to have a sense of what kind of information *can* go in (like is this just going to be "SHA-whatever was deployed to myapp-production" or does it have commit messages or tag names or what?)

JPrevost replied:
Thanks. I'll do... something here.

thatandromeda commented:
Two issues not covered in here that might be of interest:
* Static IPs
* OAuth (I have already writte OAuth documentation which I think is named "...on Heroku" but probably means "...in the cloud more generally); that doc should be either linked or incorporated here.

thatandromeda commented on readme.md:
```diff
@@ -11,3 +11,10 @@ Welcome to the MIT Libraries Developer Documentation site.
 internally facing. That said, feel free to open an Issue if you have suggestions
 or feedback.
+
+## Side Nav Ordering
```
👍  Thanks - this was not clear to me when I interacted with this system previously and it is nice to have clarification.

JPrevost replied:
It wasn't entirely clear to me and I write that script that does it. 👻

JPrevost commented:
I'll link the OAuth doc (and give it context) and mention Static IPs being not-the-norm.

Thanks for the review!

JPrevost committed:
PR feedback

- Adds mit auth info
- Adds static ip info
- Adds notes about storing ENV in LastPass
- Clarifies deploy hook audience
